src/tests/buildings/green.ts

```diff
@@ -0,0 +1,113 @@
+import { expect } from "chai";
+import { GameEngine } from "../../framework/engine";
+import { PlayerState } from "../../framework/types";
+import { requireActivePlayer } from "../../game/helpers";
+import {
+  P1,
+  debugAction,
+  debugGotoNextTurn,
+  debugPlayCard,
+  debugPlayUnit,
+  makeDefaultGame,
+} from "../testhelper";
+
+describe("buildings", () => {
+  describe("Moss Sentinels", () => {
+    let $: GameEngine;
+    let P: PlayerState;
+
+    beforeEach(() => {
+      $ = makeDefaultGame();
+      P = requireActivePlayer($);
+    });
+
+    describe("Verdant Tree", () => {
+      it.skip("should apply to existing tech buildings", () => {
+        P.workers = 6;
+
+        debugAction($, {
+          type: "PURCHASE_TECH_BUILDING",
+        });
+
+        expect(
+          () => debugPlayCard($, "Gemscout Owl")
+        ).to.throw("tech building is still being built");
+
+        const tree = debugPlayUnit($, "Verdant Tree");
+        tree.arrivalFatigue = false;
+
+        debugAction($, {
+          type: "ACTIVATE_ABILITY",
+          instance: tree.id,
+          ability: `${tree.card} #1`,
+        });
+
+        // debugValidateEffects($);
+
+        expect(() => debugPlayCard($, "Gemscout Owl")).to.not.throw();
+      });
+
+      it.skip(
+        "should apply to tech buildings purchased after its ability has " +
+          "been used",
+        () => {
+          P.workers = 6;
+
+          const tree = debugPlayUnit($, "Verdant Tree");
+          tree.arrivalFatigue = false;
+
+          expect(
+            () => debugPlayCard($, "Gemscout Owl")
+          ).to.throw("missing tech building");
+
+          debugAction($, {
+            type: "ACTIVATE_ABILITY",
+            instance: tree.id,
+            ability: `${tree.card} #1`,
+          });
+
+          // debugValidateEffects($);
+
+          debugAction($, {
+            type: "PURCHASE_TECH_BUILDING",
+          });
+
+          expect(() => debugPlayCard($, "Gemscout Owl")).to.not.throw();
+        }
+      );
+
+      it.skip(
+        "should continue to apply even if it is removed from play",
+        () => {
+          expect(false).to.equal(true);
+
+          /**
+           * There has been a nontrivial amount of discussion about this case.
```

Comment:
Ah, I think this is what I was referring to with the comment where the effect is actually implemented.

Reply:
Oh, good catch.